services/tasks-svc/main.go

```diff
@@ -14,14 +17,27 @@ const ServiceName = "tasks-svc"
 var Version string
 
 func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+
 	common.Setup(ServiceName, Version, true)
 
 	eventStore := eventstoredb.SetupEventStoreByEnv()
 	aggregateStore := eventstoredb.NewAggregateStore(eventStore)
 
-	common.StartNewGRPCServer(common.ResolveAddrFromEnv(), func(server *daprd.Server) {
+	echoProjection := echo.NewProjection(eventStore)
+	go func() {
+		if err := echoProjection.Subscribe(ctx); err != nil {
+			log.Err(err).Msg("error during echo subscription")
+			cancel()
+		}
+	}()
+
+	common.StartNewGRPCServer(context.Background(), common.ResolveAddrFromEnv(), func(server *daprd.Server) {
```

Comment:
We don't share the same context here.

This means the service will continue to run the grpc server when the subscription of the projection fails.
I don't think this is a good idea, the service should either run, and then do so properly, or fail at startup.

Btw we are good the other way around I think.

Reply:
Yes, you are right. This was my intention with the refactor of `StartNewGRPCServer()` ...